src/OrbitGl/App.cpp

```diff
@@ -387,6 +389,10 @@ void OrbitApp::OnCaptureStarted(const orbit_grpc_protos::CaptureStarted& capture
                                            absl::ZeroDuration(), warning_message);
           SendWarningToUi("Capture", warning_message);
         }
+
+        api_event_id_setter_ = orbit_capture_client::NameEqualityApiEventIdSetter::Create(
```

Comment:
As above, I think App should not know about that class. Then you also don't need to care about clearing it again.